tests/integration-tests/tests/iam/test_iam.py

```diff
@@ -332,3 +336,134 @@ def test_s3_read_write_resource(region, pcluster_config_reader, s3_bucket_factor
     # Check S3 resources
     check_s3_read_resource(region, cluster, get_policy_resources(config, enable_write_access=False))
     check_s3_read_write_resource(region, cluster, get_policy_resources(config, enable_write_access=True))
+
+
+@pytest.mark.parametrize("iam_resource_prefix", ["name-prefix-", "/path-prefix/", "/path-prefix/name-prefix-"])
+@pytest.mark.usefixtures("os", "instance")
+def test_iam_resource_prefix(
+    register_resource_prefix_cli_credentials,
+    region,
+    pcluster_config_reader,
+    clusters_factory,
+    test_datadir,
+    scheduler_commands_factory,
+    s3_bucket_factory,
+    s3_bucket,
+    iam_resource_prefix,
+):
+    bucket_name = s3_bucket
+    cfn_client, _, iam_client, _ = _create_boto3_clients(region)
+
+    create_config, update_config = _get_config_create_and_update(test_datadir)
+   
+    cluster_config = pcluster_config_reader(
+        config_file=create_config,
+        min_count=1,
+        bucket=bucket_name,
+        iam_resource_prefix=iam_resource_prefix,
+    )
+    cluster = clusters_factory(cluster_config)
+    _test_iam_resource_in_cluster(cfn_client, iam_client, cluster.name, iam_resource_prefix)
+
+def _split_resource_prefix(resource_prefix):
+    """To split Path and name prefix from Resource Prefix."""
+    if resource_prefix:
+        split_index = resource_prefix.rfind("/") + 1
+        return (
+            None
+            if split_index == 0
+            else resource_prefix
+            if split_index == len(resource_prefix)
+            else resource_prefix[:split_index],
+            None
+            if split_index == len(resource_prefix)
+            else resource_prefix
+            if split_index == 0
+            else resource_prefix[split_index:],
+        )
+    return None, None
+
+
+def _check_iam_resource_prefix(resource_arn_list, iam_resource_prefix):
+    """Check the path and name of IAM resource ( Roles, policy and Instance profiles)."""
+    iam_path, iam_name_prefix = _split_resource_prefix(iam_resource_prefix)
+    for resource in resource_arn_list:
+        if "arn:aws:iam:" in resource:
+            if iam_path:
+                assert_that(resource).contains(iam_path)
+            else:
+                assert_that(resource).contains("/parallelcluster/")
+        if iam_name_prefix:
+            assert_that(resource).contains(iam_name_prefix)
+
+
+def _test_iam_resource_in_cluster(cfn_client, iam_client, stack_name, iam_resource_prefix):
```

Comment:
Here you're verifying that the IAM resources are created with the expected ResourcePrefix.

I think that these are verifications that could and should be executed at the unit testing level, not at the integration testing level.

What do you think?

Reply:
Will check for creation of the cluster.

Reply:
Great and please remember to keep track of the necessity to add a unit tests to cover the generated resources.